README.adoc

```diff
@@ -80,25 +80,52 @@ checkout([$class: 'GitSCM',
 
 ==== Checkout with ssh and a private key credential
 
-Insert example here
+Checkout from the git client plugin source repository using https protocol, ssh private credentials, and a non-default branch. 
+The git plugin supports private key credentials provided by the link:https://plugins.jenkins.io/credentials[Jenkins credentials plugin].
+
+[source,groovy]
+---
+checkout([$class: 'GitSCM',
+          branches: [[name: 'master']],
+          userRemoteConfigs: [[credentialsId: 'e1476d45-b77d-4af6-a80f-e2fe95eca822', url: 'https://github.com/jenkinsci/git-plugin.git']]])
```

Comment:
While the credential id is valid, it is easier for readers to understand the purpose of the id if we make it more verbose.  Also need to use an ssh protocol URL in order to use an ssh private key credential.  The ssh private key credential won't work with an https URL.
```suggestion
          userRemoteConfigs: [[credentialsId: 'my-ssh-private-key-id',
                                              url: 'ssh://github.com/jenkinsci/git-plugin.git']]])
```

Reply:
Okay